Cleanup project configuration

glassez opened the pull request:
The another portion of cleanups has done while I'm updating my build environment.

Chocobo1 commented:
cmakes are still failing: https://travis-ci.org/github/qbittorrent/qBittorrent/jobs/682319805

glassez commented:
>cmakes are still failing

I fixed an existing error, as well as a regression of my previous PR that I made, confused by the first one. @Kolcha fixed another regression in a subsequent PR.

jagannatharjun commented:
https://github.com/qbittorrent/qBittorrent/blob/4e7ff88cf4e87eb847dbbae4806e9922a842d535/CMakeLists.txt#L65
I think you removed this feature

glassez commented:
>I think you removed this feature

Don't understand you...

jagannatharjun commented:
sorry I got confused between feature name and corresponding define

Chocobo1 commented:
@glassez 
#12735 is merged now, please rebase this PR. I want to make sure that cmake builds are working properly.

NotTsunami commented:
Very nice changes, the last few patchsets have been vast improvements to our CMake configuration, I actually had a commit similar to your `Don't use subprojects in GUI subfolders` commit nuking the extra CMakeLists files on my local branch. Thank you for the PRs!

glassez commented:
>#12735 is merged now

At the end I fixed this issue in another way (I fixed the cause instead of the symptoms). Otherwise it's confusing that headers from the same directory cannot be included using relative path.
